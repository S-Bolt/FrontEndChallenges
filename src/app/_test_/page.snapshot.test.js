import { render } from "@testing-library/react";
import Home from "../page"; // adjust the import path as needed

describe("Home Page Snapshot", () => {
  it("matches the snapshot", () => {
    const { asFragment } = render(<Home />);
    expect(asFragment()).toMatchSnapshot();
  });
});
