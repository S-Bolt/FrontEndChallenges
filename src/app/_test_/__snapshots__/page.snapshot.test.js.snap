// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Snapshot matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="theme-project1 bg-neutral-white text-neutral-slate-900"
    >
      <h2
        class="font-Figtree flex h-24 items-center justify-center bg-black text-center text-white"
      >
        Challenge 1
      </h2>
      <div>
        <div
          class="theme-project1 my-12"
        >
          <div
            class="grid w-full gap-8 md:grid-cols-2"
          >
            <div
              class="row-span-1 flex flex-col rounded-tr-[3rem] bg-slate-100 px-[5rem] pt-6 text-left"
            >
              <div
                class="-ml-12 -mt-[6rem]"
              >
                <img
                  alt="Favicon"
                  class="mb-4"
                  data-nimg="1"
                  decoding="async"
                  height="150"
                  loading="lazy"
                  src="/_next/image?url=%2FGroup357.png&w=384&q=75"
                  srcset="/_next/image?url=%2FGroup357.png&w=256&q=75 1x, /_next/image?url=%2FGroup357.png&w=384&q=75 2x"
                  style="color: transparent; box-shadow: none;"
                  width="150"
                />
              </div>
              <div>
                <h1
                  class="mt-12 text-3xl text-primary"
                >
                  QR code component
                </h1>
                <p
                  class="my-6"
                >
                  This is a Figma design file or a Frontend Mentor challenge. Figma is a design tool professional teams use to collaborate on projects. Need help using Figma? 
                  <a
                    class="font-bold"
                    href="https://www.frontendmentor.io/articles/figma-for-developers-how-to-work-with-a-design-file-m6CZKZ1rC1"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Read our Figma for developers article.
                  </a>
                </p>
              </div>
              <div>
                <div
                  class="mt-6 flex h-96 items-center justify-center rounded-t-lg border-l-[12px] border-r-[12px] border-t-[16px] border-black bg-neutral-slate-300"
                >
                  <div
                    class="max-w-64 scale-50 rounded-lg bg-neutral-white p-4 shadow-md"
                  >
                    <img
                      alt="QR CODE"
                      class="mx-auto mb-4"
                      data-nimg="1"
                      decoding="async"
                      height="200"
                      loading="lazy"
                      src="/_next/image?url=%2Fimage-qr-code.png&w=640&q=75"
                      srcset="/_next/image?url=%2Fimage-qr-code.png&w=256&q=75 1x, /_next/image?url=%2Fimage-qr-code.png&w=640&q=75 2x"
                      style="color: transparent;"
                      width="200"
                    />
                    <h2
                      class="text-base font-bold"
                    >
                      Improve your front-end skills by building projects
                    </h2>
                    <p
                      class="text-neutral-slate-600 mt-2 text-sm"
                    >
                      Scan the QR code to visit Frontend Mentor and take your coding skills to the next level
                    </p>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="flex flex-col justify-center px-[5rem]"
            >
              <div
                class="mb-4 flex items-start"
              >
                <div
                  class="w-1/3 p-6"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-gears mr-4 h-full w-full"
                    data-icon="gears"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 640 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M308.5 135.3c7.1-6.3 9.9-16.2 6.2-25c-2.3-5.3-4.8-10.5-7.6-15.5L304 89.4c-3-5-6.3-9.9-9.8-14.6c-5.7-7.6-15.7-10.1-24.7-7.1l-28.2 9.3c-10.7-8.8-23-16-36.2-20.9L199 27.1c-1.9-9.3-9.1-16.7-18.5-17.8C173.9 8.4 167.2 8 160.4 8l-.7 0c-6.8 0-13.5 .4-20.1 1.2c-9.4 1.1-16.6 8.6-18.5 17.8L115 56.1c-13.3 5-25.5 12.1-36.2 20.9L50.5 67.8c-9-3-19-.5-24.7 7.1c-3.5 4.7-6.8 9.6-9.9 14.6l-3 5.3c-2.8 5-5.3 10.2-7.6 15.6c-3.7 8.7-.9 18.6 6.2 25l22.2 19.8C32.6 161.9 32 168.9 32 176s.6 14.1 1.7 20.9L11.5 216.7c-7.1 6.3-9.9 16.2-6.2 25c2.3 5.3 4.8 10.5 7.6 15.6l3 5.2c3 5.1 6.3 9.9 9.9 14.6c5.7 7.6 15.7 10.1 24.7 7.1l28.2-9.3c10.7 8.8 23 16 36.2 20.9l6.1 29.1c1.9 9.3 9.1 16.7 18.5 17.8c6.7 .8 13.5 1.2 20.4 1.2s13.7-.4 20.4-1.2c9.4-1.1 16.6-8.6 18.5-17.8l6.1-29.1c13.3-5 25.5-12.1 36.2-20.9l28.2 9.3c9 3 19 .5 24.7-7.1c3.5-4.7 6.8-9.5 9.8-14.6l3.1-5.4c2.8-5 5.3-10.2 7.6-15.5c3.7-8.7 .9-18.6-6.2-25l-22.2-19.8c1.1-6.8 1.7-13.8 1.7-20.9s-.6-14.1-1.7-20.9l22.2-19.8zM112 176a48 48 0 1 1 96 0 48 48 0 1 1 -96 0zM504.7 500.5c6.3 7.1 16.2 9.9 25 6.2c5.3-2.3 10.5-4.8 15.5-7.6l5.4-3.1c5-3 9.9-6.3 14.6-9.8c7.6-5.7 10.1-15.7 7.1-24.7l-9.3-28.2c8.8-10.7 16-23 20.9-36.2l29.1-6.1c9.3-1.9 16.7-9.1 17.8-18.5c.8-6.7 1.2-13.5 1.2-20.4s-.4-13.7-1.2-20.4c-1.1-9.4-8.6-16.6-17.8-18.5L583.9 307c-5-13.3-12.1-25.5-20.9-36.2l9.3-28.2c3-9 .5-19-7.1-24.7c-4.7-3.5-9.6-6.8-14.6-9.9l-5.3-3c-5-2.8-10.2-5.3-15.6-7.6c-8.7-3.7-18.6-.9-25 6.2l-19.8 22.2c-6.8-1.1-13.8-1.7-20.9-1.7s-14.1 .6-20.9 1.7l-19.8-22.2c-6.3-7.1-16.2-9.9-25-6.2c-5.3 2.3-10.5 4.8-15.6 7.6l-5.2 3c-5.1 3-9.9 6.3-14.6 9.9c-7.6 5.7-10.1 15.7-7.1 24.7l9.3 28.2c-8.8 10.7-16 23-20.9 36.2L315.1 313c-9.3 1.9-16.7 9.1-17.8 18.5c-.8 6.7-1.2 13.5-1.2 20.4s.4 13.7 1.2 20.4c1.1 9.4 8.6 16.6 17.8 18.5l29.1 6.1c5 13.3 12.1 25.5 20.9 36.2l-9.3 28.2c-3 9-.5 19 7.1 24.7c4.7 3.5 9.5 6.8 14.6 9.8l5.4 3.1c5 2.8 10.2 5.3 15.5 7.6c8.7 3.7 18.6 .9 25-6.2l19.8-22.2c6.8 1.1 13.8 1.7 20.9 1.7s14.1-.6 20.9-1.7l19.8 22.2zM464 304a48 48 0 1 1 0 96 48 48 0 1 1 0-96z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="w-2/3"
                >
                  <h2
                    class="font-bold"
                  >
                    Design Systems
                  </h2>
                  <p>
                    The design system contains all the information for reusable components and styles. It shows colors, typography styles, and components, including various component states if they're interactive.
                  </p>
                </div>
              </div>
              <div
                class="my-6 border-4"
              />
              <div
                class="mb-4 flex items-start"
              >
                <div
                  class="w-1/3 p-6"
                >
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-palette mr-4 h-full w-full"
                    data-icon="palette"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M512 256c0 .9 0 1.8 0 2.7c-.4 36.5-33.6 61.3-70.1 61.3L344 320c-26.5 0-48 21.5-48 48c0 3.4 .4 6.7 1 9.9c2.1 10.2 6.5 20 10.8 29.9c6.1 13.8 12.1 27.5 12.1 42c0 31.8-21.6 60.7-53.4 62c-3.5 .1-7 .2-10.6 .2C114.6 512 0 397.4 0 256S114.6 0 256 0S512 114.6 512 256zM128 288a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zm0-96a32 32 0 1 0 0-64 32 32 0 1 0 0 64zM288 96a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zm96 96a32 32 0 1 0 0-64 32 32 0 1 0 0 64z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
                <div
                  class="w-2/3"
                >
                  <h2
                    class="font-bold"
                  >
                    Designs
                  </h2>
                  <p>
                    The high-fidelity designs help you build responsive, accessible projects. Aim for precision rather than pixel perfect replicas. For guidance, see Josh Corneau's 
                    <a
                      class="font-bold"
                      href="https://www.joshwcomeau.com/css/pixel-perfection/"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      "Chasing the Pixel-Perfect-Dream".
                    </a>
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="theme-project2 bg-neutral-white text-neutral-slate-900"
    >
      <h2
        class="font-Figtree flex h-24 items-center justify-center bg-black text-center text-white"
      >
        Challenge 2
      </h2>
      <div>
        <div
          class="flex h-[50rem] items-center justify-center bg-neutral-slate-100 md:h-[52rem]"
        >
          <div
            class="flex h-full w-full flex-col self-center rounded-3xl bg-neutral-white md:h-2/3 md:w-3/4 md:flex-row"
          >
            <div
              class="h-2/3 w-full p-6 md:h-full md:w-1/2 md:p-8 lg:p-10"
            >
              <div
                class="mb-1 md:flex md:flex-row md:items-center md:justify-between lg:mb-2"
              >
                <h2
                  class="font-family-Jakarta text-xl font-bold text-neutral-slate-900 md:text-2xl lg:mb-2"
                >
                  Mortgage Calculator
                </h2>
                <button
                  class="text-neutral-slate-500 underline"
                >
                  Clear All
                </button>
              </div>
              <div
                class="lg:mb-2"
              >
                <h3
                  class="mb-1 lg:mb-2"
                >
                  Mortgage Amount
                </h3>
                <div
                  class="group flex items-center"
                >
                  <label
                    class="rounded-l-md border border-r-0 border-neutral-slate-500 bg-neutral-slate-100 px-3 py-1 group-focus-within:border-primary group-focus-within:bg-primary"
                    for="mortgageAmount"
                  >
                    $
                  </label>
                  <input
                    class="font-family-Jakarta focus-bg-primary h-[2.15rem] w-full rounded-r-md border border-l-0 border-neutral-slate-500 pl-2 font-bold focus:border-primary focus:outline-none"
                    id="mortgageAmount"
                    type="text"
                    value=""
                  />
                </div>
                <div
                  class="mt-1 h-5"
                />
              </div>
              <div
                class="flex flex-col lg:mb-2 lg:flex-row lg:space-x-6"
              >
                <div>
                  <h3
                    class="lg:mb-2"
                  >
                    Mortgage Term
                  </h3>
                  <div
                    class="group flex items-center md:mb-1 lg:mb-0"
                  >
                    <input
                      class="font-family-Jakarta h-[2.15rem] w-full rounded-l-md border border-r-0 border-neutral-slate-500 pl-2 font-bold focus:border-primary focus:outline-none"
                      id="termInput"
                      type="text"
                      value=""
                    />
                    <label
                      class="rounded-r-md border border-l-0 border-neutral-slate-500 bg-neutral-slate-100 px-3 py-1 group-focus-within:border-primary group-focus-within:bg-primary"
                      for="termInput"
                    >
                      Years
                    </label>
                  </div>
                  <div
                    class="mt-1 h-5"
                  />
                </div>
                <div>
                  <h3
                    class="lg:mb-2"
                  >
                    Interest Rate
                  </h3>
                  <div
                    class="group flex items-center"
                  >
                    <input
                      class="font-family-Jakarta h-[2.15rem] w-full rounded-l-md border border-r-0 border-neutral-slate-500 pl-2 font-bold focus:border-primary focus:outline-none"
                      id="rateInput"
                      type="text"
                      value=""
                    />
                    <label
                      class="rounded-r-md border border-l-0 border-neutral-slate-500 bg-neutral-slate-100 px-3 py-1 group-focus-within:border-primary group-focus-within:bg-primary"
                      for="rateInput"
                    >
                      %
                    </label>
                  </div>
                  <div
                    class="mt-1 h-5"
                  />
                </div>
              </div>
              <div>
                <h3
                  class="lg:mb-2"
                >
                  Mortgage Type
                </h3>
                <div
                  class="flex flex-col md:mb-1"
                >
                  <label
                    class="mb-3 flex h-[2.15rem] w-full cursor-pointer items-center space-x-4 border font-bold border-neutral-slate-500"
                    for="repayment"
                  >
                    <input
                      class="ml-4 accent-white"
                      data-testid="repayment-radio"
                      id="repayment"
                      name="mortgageType"
                      type="radio"
                      value="repayment"
                    />
                    <span
                      class="font-family-Jakarta"
                    >
                      Repayment
                    </span>
                  </label>
                  <label
                    class="mb-1 flex h-[2.15rem] w-full cursor-pointer items-center space-x-4 border font-bold lg:mb-2 border-neutral-slate-500"
                  >
                    <input
                      class="ml-4 accent-white"
                      name="mortgageType"
                      type="radio"
                      value="interest-only"
                    />
                    <span
                      class="font-family-Jakarta font-bold"
                    >
                      Interest Only
                    </span>
                  </label>
                  <div
                    class="mt-1 h-5"
                  />
                </div>
              </div>
              <button
                class="h-12 w-full rounded-3xl bg-primary lg:w-4/5"
              >
                <div
                  class="flex items-center justify-center"
                >
                  <img
                    alt="Calculator Icon"
                    class="mx-2"
                    data-nimg="1"
                    decoding="async"
                    height="20"
                    loading="lazy"
                    src="/icon-calculator.svg"
                    style="color: transparent;"
                    width="20"
                  />
                  <span
                    class="font-bold"
                  >
                    Calculate Repayment
                  </span>
                </div>
              </button>
            </div>
            <div
              class="flex h-1/3 w-full flex-col items-center justify-center bg-neutral-slate-900 p-2 md:h-full md:w-1/2 md:rounded-bl-[3.5rem] md:rounded-br-3xl md:rounded-tr-3xl md:p-8 lg:p-10"
              data-testid="results-container"
            >
              <img
                alt="Calculator Icon"
                class="mx-2 mt-6 h-32 w-32 md:mt-0 md:h-48 md:w-48"
                data-nimg="1"
                decoding="async"
                height="200"
                loading="lazy"
                src="/illustration-empty.svg"
                style="color: transparent;"
                width="200"
              />
              <h3
                class="font-family-Jakarta mb-2 text-xl font-semibold text-neutral-white md:text-2xl"
              >
                Results shown here
              </h3>
              <p
                class="font-family-Jakarta mb-8 text-center text-sm font-semibold text-neutral-slate-500 md:text-base"
              >
                Complet the form and click "calculate repayments" to see weat your monthly repayments would be.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
